handlers/eth1Transactions.go

```diff
@@ -146,6 +146,10 @@ func getEth1BlockAndNext(number uint64) (*types.Eth1Block, uint64, error) {
 		return nil, 0, fmt.Errorf("block %d not found", number)
 	}
 
+	if number == 0 {
+		return block, 0, nil
```

Comment:
No, 0 is the first block and there is no next as we are sorting in descending order.

Reply:
Because this is exactly what would be returned without the early exit too. This is NOT a behavior change, it is just faster now.
We could discuss whether it makes sense to return a number as `nextBlock` that is a valid block number even when there is no valid next block.

My stance: No, this does not make sense.
Also my stance: Changing this is out of scope of this PR.

What are your thoughts?

Reply:
Was not aware that this fetches in descending order. In that case, `getEth1BlockAndNext` is a misleading function name, since it lead me to believe it would return the block and it's following block in the canonical blockchain. (If you don't even wan't to change the name resolve this convo)

After looking at it for a while, this whole function seems weird? Two BT fetches that include the same block?? The "next" block is only fetched for it's block number even though it can't be anything else than the first block's number minus one???

Reply:
I have at least improved the comment for the function, see 06f8df19707d11ee4f0fd7ec9acedfea26f8375d

As you might know, I did not implement the function but I saw a potential early exit that removes a bigtable call so I added it. Without spending too much time on pondering the function, the two calls most likely come from the following problematic:

We don't know what block exactly comes after the one requested in bigtable. We make one call to check that the block we want is there and then another one to get the one after it. As we don't know what block is after the given block, we have to query the first one again in the second query.
Something like that.

I believe that this function can be improved but I feel like that's out of scope for this PR. Feel free to create an internal ticket for this topic.